birdcage_backend/start_app.sh

```diff
@@ -1,8 +1,28 @@
 #!/bin/bash
 
+#
+# Gunicorn won't pass environment variables by default,
+# so pass anything we might be specifying in docker-compose.
+#
+# SCRIPT_NAME is a wsgi convenience feature that gunicorn
+# uses to tell Flask that the application may live in a subfolder.
+#
+
 # Start the Flask app using gunicorn
 echo "Starting API Server"
-gunicorn --bind 0.0.0.0:$API_SERVER_PORT run:app > >(cat >&1) 2> >(cat >&2) &
+gunicorn --bind 0.0.0.0:${API_SERVER_PORT} \
+	-e ANALYZE_PORT="${ANALYZE_PORT}" \
+	-e ANALYZE_SERVER="${ANALYZE_SERVER}" \
+	-e CONCURRENCY="${CONCURRENCY}" \
+	-e CORS_ORIGINS="${CORS_ORIGINS}" \
+	-e DATABASE_FILE="${DATABASE_FILE}" \
+	-e DETECTION_DIR_NAME="${DETECTION_DIR_NAME}" \
+	-e JWT_SECRET_KEY="${JWT_SECRET_KEY}" \
+	-e REDIS_PORT="${REDIS_PORT}" \
+	-e REDIS_SERVER="${REDIS_SERVER}" \
+	-e SCRIPT_NAME="${SCRIPT_NAME}" \
+	-e TEMP_DIR_NAME="${TEMP_DIR_NAME}" \
+        run:app > >(cat >&1) 2> >(cat >&2) &
```

Comment:
Not sure if you ever noticed this, but none of the environment variables specified in docker-compose actually worked, because apparently Gunicorn doesn't pass the environment into the running context when it starts if you don't explicitly pass variables. There may be a cleaner way to do this..

Reply:
I think it is fair to say that I never noticed that. I'd have guessed that would have become obvious at some point.